lib/facter/mongodb_version.rb

```diff
@@ -1,8 +1,8 @@
 Facter.add(:mongodb_version) do
+  confine { Facter::Core::Execution.which('mongo') }
```

Comment:
https://puppet.com/docs/facter/3.11/custom_facts.html#confining-facts doesn't mention this form at all. I'm hesitant to rely on something that doesn't appear to be what it was intended to do.